### PR TITLE
feat(manager/cargo): support dynamic install 🎉

### LINKED ISSUE #18661: Support binarySource=install for rust
### What would you like Renovate to be able to do?

Support dynamic installation of rust when calling cargo

### If you have any ideas on how this should be implemented, please tell us here.

Usual

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by viceice**
## Changes
Support dynamic rust install 🎉

## Context
- closes #18661

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
waiting for sidecar update

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -175,6 +175,7 @@ Supported tools for dynamic install are:
 - `pnpm`
 - `poetry`
 - `python`
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:177]: Will `semantic-release` work properly if the PR title has an emoji like 🎉?

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:177]: Nevermind, I see `semantic-release` handles emojis just fine. 🎉 😏

Comment (viceice) [reply docs/usage/self-hosted-configuration.md:177]: 😜

**Review by rarkins — APPROVED** [review]

**Pull request merged**